galaxy_skeleton2.R

```diff
@@ -79,28 +80,52 @@ variable_names <-
 # super for this kind of search-and-replace. Replace the question mark below.
 
 comma_separated_values <- 
-  ? %>% 
+  raw_file[L+1:810] %>% 
   gsub("\\|", ",", .) %>% 
   gsub(" ", "", .)
-
+comma_separated_values
 # We then just add the variable names (separated with commas) on top, and
 # cat()-the whole ting to a .csv-file in the same way as we did with the
 # variable descriptions above.
 
 comma_separated_values_with_names <- 
   c(paste(variable_names, collapse = ","),
     comma_separated_values)    
-
+comma_separated_values_with_names
 # Replace the question mark and come up with a file name
-cat(?, sep = "\n", file = "?")
+cat(comma_separated_values_with_names, sep = "\n", file = "variable_names_sep_comma.CSV")
 
 # Read the file back in as a normal csv-file. The readr-package is part of
 # tidyverse, so it is already loaded.
-galaxies <- read_csv("?")
-
+galaxies <- read_csv("variable_names_sep_comma.CSV")
+#Remove empty lines/rows.
+galaxies <- subset(galaxies, subset = !is.na(galaxies$name))
 
 # You should now have a nice, clean data frame with galaxies and their
 # characteristics in memory. As of March 2022 it should contain 796
 # observations.
 
-
+#Problem 3----
+#I create a new column to specify whether the galaxy is big or small
+# i categorize the galaxies by saying small galaxies has a a_26 less than 1 
+#and the rest are big
+galaxies <- galaxies %>% 
+  mutate(size_category = if_else(a_26 < 1 , "small", "big"))
+
+#Make a plot to vizualize the frequenzy of big and small galaxies:
+galaxies %>% 
+  ggplot(aes(x = size_category)) +
+  geom_bar() +
+  labs(x = "Size Category",
+       y = "Frequency",
+       title = "Frequency of Size Categories") +
+  theme_minimal()
```

Comment:
Nice plot. However, it is hard to know which threshold you should use to classify what is big and small objects. Perhaps it is better with a scatterplot against distance? Then you would see that the smallest observations are close to us ('D' is low).